pkg/providers/gcp/provider_postgres.go

```diff
@@ -203,6 +211,18 @@ func (p *PostgresProvider) reconcileCloudSQLInstance(ctx context.Context, pg *v1
 				msg := fmt.Sprintf("failed to modify cloudsql instance: %s", foundInstance.Name)
 				return nil, croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
 			}
+
+			// Disable Maintenance - set Deny
+			logger.Infof("disable Maintenance - set Deny maintenance period 90 days (max allowed by GCP)")
+			modifiedInstance.Settings.DenyMaintenancePeriods = []*sqladmin.DenyMaintenancePeriod{{
+				EndDate:   time.Now().Add(time.Hour * 2160).Format("01-02-2006"),
+				StartDate: time.Now().Format("01-02-2006"),
+			}}
+			_, err = sqladminService.ModifyInstance(ctx, strategyConfig.ProjectID, foundInstance.Name, modifiedInstance)
```

Comment:
I think that there is a chance that we remove the deny period and apply it again before the maintenance is scheduled so I'm not sure if this will work as intended. I think we need to see a maintenance event so that we know how to determine whether the maintenance has been performed or not.

Reply:
Thank you. If I understand the comment, -  after the Maintenance update applying  and `sqladminService.ModifyInstance(.....)` , need `WAIT` before set Disable update (and modify instance again).
Yes, I thought, but in current scenario - the second modification should not impact, as maintenance update already started. If it's not succeed, -  we will have another/next `maintenanceWindowEnabled` chance. Maybe small sleep, but seems to me no need  check/wait if maintenance update succeed.

Reply:
My concern is that it will remove the deny from the instance and then apply it again without the maintenance taking place. Normally once CRO successfully reconciles in this case it removes the maintenaceWindow flag from the Postgres CR so the update might not take place until the next service affecting RHOAM release

Reply:
agree, maybe risk of such scenario exists, as you noted "remove deny, update not stated yet, apply deny". So update will not happen. Don't know ... maybe sleep (although I also not so like it, but meanwhile and open case to improve it), but not wait for update completion; maybe it will take hours. Seems need somehow check if maintenance upgrade started. Maybe another ticket with exact problem description.

Reply:
Yep I think perhaps we opt to just extend the maintenance deny for now and do not remove it - we can have a follow up ticket that looks into actually performing the update once we have an affected cluster where we can test the issue.

Reply:
sorry, if do not remove Deny - maintenance update is not possible, as denied (it's differ from AWS).  Seems better as is (+ small sleep), as seems to me.

Reply:
Sorry if I'm misremembering but the suggestion we had on a call is to update the deny period continuously for now so that no service maintenance takes place in the first GCP CRO release. Once we get a new maintenance event it will be denied until we implement a solution that we know is working (as we will be able to test it) and we will release a new CRO version and RHOAM release to roll out the maintenance. In my opinion this is a safer approach that gives us time to get the solution working correctly.

Reply:
This is exactly what we have now. Maintenance will never happen if `maintenanceWindowEnabled  is false.
In this case the flow is go to `checkUpdateDenyPeriod`. And here - we continuously update the deny period.

Reply:
That's correct but if there is a serviceAffecting RHOAM release where we have not had time to fix the issue then these untested changes will run. If you really are adamant to keep this in then we can but in my opinion we should only have tested working changes in the releases.

Reply:
Thank you @adam-cattermole , maybe we could discuss also in GCP chat, how to progress with this. Any decision is fine for me. Of cause that not verified things should NOT be included into Release. But if feature is Needed, it should be verified.

Reply:
Hi @adam-cattermole , postgres maintenance, Deny set/update seems working stable. I added handling of the date format issue.